src/components/Pagination/PaginationLink.tsx

```diff
@@ -1,36 +1,48 @@
 /** @jsx h */
 
 import { h } from 'preact';
+import cx from 'classnames';
 
-export type PaginationLinkComponentCSSClasses = {
-  item: string;
-  link: string;
-};
+import type { PaginationComponentCSSClasses } from './Pagination';
 
-export type PaginationLinkProps = {
-  ariaLabel: string;
-  cssClasses: PaginationLinkComponentCSSClasses;
-  handleClick(pageNumber: number, event: MouseEvent): void;
-  isDisabled: boolean;
+type PageLinkProps = {
   label: string;
+  ariaLabel: string;
   pageNumber: number;
-  url?: string;
+  additionalClassName: string | null;
+  isDisabled?: boolean;
+  isSelected?: boolean;
+  cssClasses: PaginationComponentCSSClasses;
+  createURL(value: number): string;
+  handleClick(pageNumber: number, event: MouseEvent): void;
 };
 
-const PaginationLink = ({
-  cssClasses,
+function PaginationLink({
   label,
   ariaLabel,
-  url,
-  isDisabled,
-  handleClick,
   pageNumber,
-}: PaginationLinkProps) => {
+  additionalClassName = null,
+  isDisabled = false,
+  isSelected = false,
+  cssClasses,
+  createURL,
+  handleClick,
+}: PageLinkProps) {
+  const classes = {
+    item: cx(
+      cssClasses.item,
+      additionalClassName,
+      isDisabled && cssClasses.disabledItem,
+      isSelected && cssClasses.selectedItem
+    ),
+    link: cssClasses.link,
+  };
```

Comment:
We could reduce nesting by having two variables: `itemClassName` and `linkClassName`. The object as a whole doesn't hold any specific meaning.